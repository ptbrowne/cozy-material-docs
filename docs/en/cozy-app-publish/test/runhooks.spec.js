/* eslint-env jest */
const runHooks = require('../utils/runhooks')

describe('RunHooks', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const optionsMock = {
    appBuildUrl: 'http://example.mock',
    appSlug: 'cozy-mock',
    appType: 'webapp',
    appVersion: '0.0.1',
    registryUrl: 'http://registry.mock/',
    registryEditor: 'Cozy',
    registryToken: 'a5464f54e654c6546b54a56a'
  }

  it('handles hook', async () => {
    await expect(
      runHooks('./test/__mocks__/hook', 'pre', optionsMock)
    ).resolves.toMatchSnapshot()
  })

  it('throws error on invalid prepublish hook path', async () => {
    await expect(
      runHooks('./test/__mocks__/not-existing-hook', 'post', optionsMock)
    ).rejects.toMatchSnapshot()
  })

  it('throws error when prepublish hook throws error', async () => {
    await expect(
      runHooks('./test/__mocks__/errored-hook', 'pre', optionsMock)
    ).rejects.toMatchSnapshot()
  })
})
