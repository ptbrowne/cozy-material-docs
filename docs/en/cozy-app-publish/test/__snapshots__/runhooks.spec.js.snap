// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunHooks handles hook 1`] = `
Object {
  "appBuildUrl": "http://cozy.io",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
}
`;

exports[`RunHooks throws error on invalid prepublish hook path 1`] = `[Error: postpublish hooks failed]`;

exports[`RunHooks throws error when prepublish hook throws error 1`] = `[Error: prepublish hooks failed]`;
